### PR TITLE
[DropdownMenu][ContextMenu] Improve `SubContent` types

### PR DESCRIPTION
**Pull request by andy-hook**
- Tidies up nesting logic
- Exposes `SubContent` part with accurate prop typings

I haven't re-ordered anything in the file yet to make the diff easier to read, not sure if we think it would be best to move them to the bottom of the file as well.

### CODE REVIEW

File: packages/react/menu/src/Menu.tsx [hunk 1]
```diff
@@ -165,21 +164,24 @@ const MenuSub: React.FC<MenuSubProps> = (props: ScopedProps<MenuSubProps>) => {
...
         open={open}
         onOpenChange={handleOpenChange}
         content={content}
         onContentChange={setContent}
         onRootClose={parentMenuContext.onRootClose}
-        contentId={useId()}
-        trigger={trigger}
-        onTriggerChange={setTrigger}
-        triggerId={useId()}
         modal={false}
```
Comment (benoitgrelard) [inline packages/react/menu/src/Menu.tsx:174]: This isn't used by the submenu right? Do you think it's worth splitting out a general menu context from a root context / sub context?

Not sure if there are other things like this one which only are used by the root?

File: packages/react/menu/src/Menu.tsx [hunk 2]
```diff
@@ -1260,6 +1257,7 @@ const Sub = MenuSub;
 const Anchor = MenuAnchor;
 const SubTrigger = MenuSubTrigger;
 const Content = MenuContent;
+const SubContent = MenuSubContent;
```
[the remark this answers is not in the document]
Comment (benoitgrelard) [reply packages/react/menu/src/Menu.tsx:1262]: I think we should mirror what we did in the other 2, so put the sub stuff at the bottom. Thanks for not doing it straight away though, it's already pretty tricky to review this PR 😅

**Review by benoitgrelard — APPROVED** [review]
I think this is all correct 😅, pretty hard to follow the changes I have to admit though. Just the nature of it.

Left a few minor comments inline.

File: packages/react/menu/src/Menu.tsx (hunk 1 above)
Comment (andy-hook) [reply packages/react/menu/src/Menu.tsx]: That’s a good point, in fact I don’t think we even need to recursively drill some of this stuff down anymore, we can just get it from root directly now.

  How about this? https://github.com/radix-ui/primitives/pull/1406/commits/13749a630b565c468cd604c6ddbb6120e60f5d72

Comment (benoitgrelard) [reply packages/react/menu/src/Menu.tsx]: I'm not sure about this. I feel like it should be the other way around, ie:
- `MenuContext` contains the things that are needed/used by the entire tree
- `MenuRootContext` contains the things that are only needed by the root
- `MenuSubContext` contains the things that are only needed by a sub

So for example `modal` would be only on `MenuRootContext`, but `dir` would be on `MenuContext`.

Does that make sense to you?

Comment (andy-hook) [reply packages/react/menu/src/Menu.tsx]: Works for me, is this what you meant? https://github.com/radix-ui/primitives/pull/1406/commits/006b3109548227b1a09a38613e4a7f02355a8106
